src/AlgebraicGeometry/Schemes/Varieties/Types.jl

```diff
@@ -0,0 +1,94 @@
+@doc Markdown.doc"""
+    AbsAffineAlgebraicSet <: AbsSpec
+
+An affine, geometrically reduced scheme of finite type over a field.
+"""
+abstract type AbsAffineAlgebraicSet{BaseField<:Field, RingType} <:AbsSpec{BaseField, RingType}# where {BaseField <:Field, RingType}
+end
+
+@doc Markdown.doc"""
+    AbsProjectiveAlgebraicSet <: AbsProjectiveScheme
+
+An affine, geometrically reduced scheme of finite type over a field.
+"""
+abstract type AbsProjectiveAlgebraicSet{BaseField<:Field, RingType} <:AbsProjectiveScheme{BaseField, RingType}# where {BaseField <:Field, RingType}
+end
+
+################################################################################
+#
+# Abstract types for varieties
+#
+################################################################################
+
+@doc Markdown.doc"""
+    AbsAffineVariety <: AbsAffineAlgebraicSet
+
+An affine, geometrically integral scheme of finite type over a field.
+"""
+abstract type AbsAffineVariety{BaseField<:Field, RingType} <:AbsAffineAlgebraicSet{BaseField, RingType}# where {BaseField <:Field, RingType}
+end
+
+@doc Markdown.doc"""
+    AbsProjectiveVariety <: AbsProjectiveAlgebraicSet
+
+A projective variety over a field.
+
+That is a projective, geometrically integral scheme over a field.
+"""
+abstract type AbsProjectiveVariety{BaseField<:Field, GradedRingType<:Ring} <: AbsProjectiveAlgebraicSet{BaseField, GradedRingType} end
+
+@doc Markdown.doc"""
+    AbsCoveredVariety <: Scheme
+
+A separated, geometrically integral scheme of finite type over a field.
+
+Note that we allow reducible varieties.
+"""
```

Comment:
What is meant here: The variety is (geometrically) integral, but may be reducible?

Reply:
leftover from an old version. Thanks for catching.

Reply:
I deleted the "Note..." .